src/pages/api/modules/dns-hole/summary.ts

```diff
@@ -26,21 +30,46 @@ export const Get = async (request: NextApiRequest, response: NextApiResponse) =>
     const app = applicableApps[i];
 
     try {
-      const piHole = new PiHoleClient(app.url, findAppProperty(app, 'password'));
-
-      // eslint-disable-next-line no-await-in-loop
-      const summary = await piHole.getSummary();
-
-      data.domainsBeingBlocked += summary.domains_being_blocked;
-      data.adsBlockedToday += summary.ads_blocked_today;
-      data.dnsQueriesToday += summary.dns_queries_today;
-      data.status.push({
-        status: summary.status,
-        appId: app.id,
-      });
-      adsBlockedTodayPercentageArr.push(summary.ads_percentage_today);
+      switch (app.integration?.type) {
+        case 'pihole': {
+          const piHole = new PiHoleClient(app.url, findAppProperty(app, 'password'));
+          const summary = await piHole.getSummary();
+
+          data.domainsBeingBlocked += summary.domains_being_blocked;
+          data.adsBlockedToday += summary.ads_blocked_today;
+          data.dnsQueriesToday += summary.dns_queries_today;
+          data.status.push({
+            status: summary.status,
+            appId: app.id,
+          });
+          adsBlockedTodayPercentageArr.push(summary.ads_percentage_today);
+          break;
+        }
+        case 'adGuardHome': {
+          const adGuard = new AdGuard(
+            app.url,
+            findAppProperty(app, 'username'),
+            findAppProperty(app, 'password')
```

Comment:
Maybe you could make this function variadic in order to reduce the amount of calls to it